### ISSUE TITLE
Libraries not included

### ISSUE THREAD

**Issue by imgeorgiev** [ethz-asl/eclipsify#43]
Hello,

I am using the eclipsify to setup a ROS C++/CUDA in Eclipse Nsight (aka. Eclipse CDT with CUDA support) with the ultimate goal of debugging code on a remote machine.

I have followed the guide on the README but when I import the "eclipsified" package, it can't find any of the libraries as they aren't in the include directives. Some example libraries I am using are roscpp, opencv, eigen, custom ros messages from another package and dynamic reconfigure. Even if I try adding all of these via the project properties settings in Eclipse, I still can't get it to index the files correctly. That being said, everything builds and runs fine in Eclipse.

NB. I have tried cleaning, rebuilding and reindexing within eclipse.

**Comment by HannesSommer** [comment]
Could it be that your build output parser does not find the includes? Your project's settings would then look about that:
![output-parser-provider](https://user-images.githubusercontent.com/1921422/66541342-43809000-eb2f-11e9-86b9-cc7e400e53b9.png)
If it worked well then the marked "CDT GCC Build output parser" would have the black arrow in front and on click it would show the includes found.

If that is the case you should check your settings for it by choosing the tab "Providers" instead of "Entries".  I would guess that your compiler pattern doesn't match your compiler OR you never really managed to compile with verbose output inside excplipse. It is crucial that in the eclipse console you find compiler commands with the "-I" arguments followed by the includes. This is what the output parser looks for to find the includes.

**Comment by imgeorgiev** [comment]
You are correct, my "CDT GCC Build output parser" is empty and looks similar to your screenshot. That is easily fixable if I add a dummy cpp executable that includes all of the necessary libraries (or simply add the libraries manually).

However, that does not fix the root issue - the CUDA files and the C++ libraries they use are still not indexed correctly.

Your answer has me wondering though, maybe the build output parser is looking only for .cpp binaries matching the .gcc patterns but it fails to find the CUDA binaries, thus failing to index anything related to them?

**Comment by HannesSommer** [comment]
I don't think I fully understand your idea. AFAIK, the output parser is looking for the compiler in the command lines, e.g. /usr/bin/c++. Indeed, if CUDA introduces compilers with funny names this can be a problem. Does it?
